modules/missenaBidAdapter.js

```diff
@@ -0,0 +1,87 @@
+import * as utils from '../src/utils.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+
+const BIDDER_CODE = 'missena';
+const ENDPOINT_URL = 'https://bid.missena.io/';
+
+export const spec = {
+  aliases: [BIDDER_CODE],
+  code: BIDDER_CODE,
+  gvlid: 687,
+  supportedMediaTypes: [BANNER],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
+    return typeof bid == 'object' && !!bid.params.apiKey;
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const req = validBidRequests[0];
+    const apiKey = req.params.apiKey;
+
```

Comment:
Being picky here but shouldn't `req` and `api` key be evaluated for each valid bid in map below? Assuming there are multiple missena bids in a request...

Reply:
Thanks for catching it!